File: android-compose/app/src/main/kotlin/com/dashit/app/data/model/Category.kt
package com.dashit.app.data.model

data class Category(
    val id: String,
    val name: String,
    val icon: String? = null,
    val image: String? = null,
    val sortOrder: Int? = 0,
    val itemCount: Int? = null
)

data class CategoryTile(
    val id: String,
    val name: String,
    val previewImages: List<String>,
    val productCount: Int
)

File: ios-swift/DASHit/Models/Category.swift
import Foundation

public struct Category: Codable, Identifiable, Hashable {
    public let id: String
    public let name: String
    public let icon: String?
    public let image: String?
    public let sortOrder: Int?
    public let itemCount: Int?
    
    public init(id: String, name: String, icon: String? = nil, image: String? = nil, sortOrder: Int? = 0, itemCount: Int? = nil) {
        self.id = id
        self.name = name
        self.icon = icon
        self.image = image
        self.sortOrder = sortOrder
        self.itemCount = itemCount
    }
}
